tracy-client/src/gpu.rs

```diff
@@ -0,0 +1,201 @@
+use once_cell::sync::Lazy;
+use std::{
+    convert::TryInto,
+    sync::{
+        atomic::{AtomicU16, Ordering},
+        Arc, Mutex,
+    },
+};
+
+use crate::Client;
+
+#[repr(u8)]
+/// The API label associated with the given gpu context. The list here only includes
+/// APIs that are currently supported by Tracy's own gpu implementations.
+//
+// Copied from `tracy-client-sys/tracy/common/TracyQueue.hpp:391`. Comment on enum states
+// that the values are stable, due to potential serialization issues, so copying this enum
+// shouldn't be a problem.
+pub enum GpuContextType {
+    /// Stand in for other types of contexts.
+    Invalid = 0,
+    /// An OpenGL context
+    OpenGL = 1,
+    /// A Vulkan context
+    Vulkan = 2,
+    /// An OpenCL context
+    OpenCL = 3,
+    /// A D3D12 context.
+    Direct3D12 = 4,
+    /// A D3D11 context.
+    Direct3D11 = 5,
+}
+
+///
+#[derive(Clone)]
+pub struct GpuContext {
```

Comment:
When the `"enable"` feature is unset, this struct's definition will become public. Include a private field to mitigate this. Similar for `GpuSpan`.